lib/core/jtrace.c

```diff
@@ -550,6 +742,275 @@ j_trace_enter(gchar const* name, gchar const* format, ...)
+					if (strcmp(name, "batch_start") == 0)
+					{
+						va_start(args, format);
+						row->namespace = va_arg(args, const char*);
+						va_end(args);
+						store_kv_namespace(trace_thread, row->namespace, NULL);
+					}
+					else if (strcmp(name, "put") == 0)
+					{
+						va_start(args, format);
+						va_arg(args, void*);
+						row->name = va_arg(args, const char*);
+						va_arg(args, void*);
+						row->size = va_arg(args, guint32);
+						va_end(args);
+					}
+					else if (strcmp(name, "delete") == 0)
+					{
+						va_start(args, format);
+						va_arg(args, void*);
+						row->name = va_arg(args, const char*);
+						va_end(args);
+					}
+					else if (strcmp(name, "get") == 0)
+					{
+						va_start(args, format);
+						va_arg(args, void*);
+						row->name = va_arg(args, const char*);
+						va_arg(args, void*);
+						trace_thread->access.utility_ptr = va_arg(args, void*);
+						va_end(args);
+					}
+					else if (strcmp(name, "get_all") == 0)
+					{
+						va_start(args, format);
+						row->namespace = va_arg(args, const char*);
+						va_end(args);
+						store_kv_namespace(trace_thread, row->namespace, NULL);
+					}
+					else if (strcmp(name, "get_by_prefix") == 0)
+					{
+						va_start(args, format);
+						row->namespace = va_arg(args, const char*);
+						row->name = va_arg(args, const char*);
+						va_end(args);
+						store_kv_namespace(trace_thread, row->namespace, row->name);
+					}
+
+					// iterate, init, fini <- no further deatils
+					// batch_execute <- handlen in leave
+				}
+				else if (type == 2)
+				{
+					name += 3;
+					row->backend = "db";
+					row->operation = name;
+					row->type = trace_thread->access.db.type;
+					row->path = trace_thread->access.db.config_path;
+					row->namespace = trace_thread->access.db.namespace;
+					if (strcmp(name, "batch_start") == 0)
+					{
+						va_start(args, format);
+						row->namespace = va_arg(args, const char*);
+						va_end(args);
+						store_db_namespace(trace_thread, row->namespace);
+					}
+					else if (strcmp(name, "schema_create") == 0)
+					{
+						va_start(args, format);
+						va_arg(args, void*);
+						row->name = va_arg(args, const char*);
+						row->size = va_arg(args, guint32);
+						row->bson = va_arg(args, const bson_t*);
+						va_end(args);
+					}
+					else if (strcmp(name, "schema_get") == 0)
+					{
+						va_start(args, format);
+						va_arg(args, void*);
+						row->name = va_arg(args, const char*);
+						va_end(args);
+					}
+					else if (strcmp(name, "schema_delete") == 0)
+					{
+						va_start(args, format);
+						va_arg(args, void*);
+						row->name = va_arg(args, const char*);
+						va_end(args);
+					}
+					else if (strcmp(name, "insert") == 0)
+					{
+						va_start(args, format);
+						va_arg(args, void*);
+						row->name = va_arg(args, const char*);
+						row->size = va_arg(args, guint32);
+						row->bson = va_arg(args, const bson_t*);
+						/// \TODO complexity needed?
```

Comment:
Complexity could be nice to have since it enables statistics on typical insert sizes?